src/libraries/System.Linq/src/System/Linq/Range.cs

```diff
@@ -18,7 +18,7 @@ public static IEnumerable<int> Range(int start, int count)
 
             if (count == 0)
             {
-                return Empty<int>();
+                return [];
```

Comment:
This would also make the returned type subject to compiler implementation. Should we perhaps be more prescriptive for IEnumerable types?

Reply:
> Should we perhaps be more prescriptive for IEnumerable types?

I don't want to guarantee the type that's returned. This doesn't seem different than accepting whatever type the compiler manufactures for an iterator. I think we're fighting an uphill battle if we discourage use of []. The only places I didn't use it were in places I couldn't (ie it wouldn't compile) or in Empty itself, as I can imagine the compiler might at some point return that.